### DISCUSSION TITLE
Z3950 (Unimarc) analyzer and alternative_id_1 field (m_sru003)

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#339] [Q&A] [answered]
In  biblioteq_marc.cc file
```
void biblioteq_marc::parseBookSRUMarc21
    ...
    ...
    if(tag == "003")
    ...
    ...
    m_sru003 = str.mid(index);
```

I would like the field to be filled in for the Z3950 unimarc analyzer :

```
void biblioteq_marc::parseBookZ3950Unimarc(void)
      ...
      ...   
      else if(str.startsWith("003 "))
      {
        str = str.mid(str.indexOf("ark:/"), -1);

        m_sru003 = str;
      } 
```
      
-> unable to enter value in alternative_id_1? field (m_sru003).

if I modify to the lccontrolnumber (m_lcnum) field it works fine.

```
      else if(str.startsWith("003 "))
      {
        str = str.mid(str.indexOf("ark:/"), -1);

        m_lcnum = str;
      } 
```
     
Is there anything special about the alternative_id_1 or m_sru003 fields?

Could you enlighten me?  Thank you in advance.

**Comment by textbrowser** [comment]
Busy with other things and don't understand the request.

**Reply by meteos77** [reply]
If you don't have the time, we'll deal with that later.

**Comment by textbrowser** [comment]
Please rephrase your request.

**Comment by textbrowser** [comment]
2023.01.16.

<li>Book SRU query results: set Alternate Identifier to control field 003's ark value if Alternate Identifier is empty. <b>GitHub discussion #268.</b></li>

**Comment by textbrowser** [comment]
So as I always request, an ISBN helps.

**Comment by textbrowser** [comment]
https://github.com/textbrowser/biblioteq/commit/ce1bda628a4fd9bc87b7f8f08bde59022b0d0a57

For anyone else reading, requests without repeatability are like screaming into the void. Help me help you or continue screaming into the  void.

**Comment by textbrowser** [comment, marked as answer]
Proper requests / questions:

1. I am doing this using this ISBN or this ISSN or this whatever.
2. I am querying this server using this format or whatever.
3. My results are this and I'm confused or whatever.
4. I think my operating system is this.
5. I think I'm using BQ version whatever.
6. I care about your time so let me help you help me.

Don't:
1. Help me I'm confused and I need attention.

**Comment by meteos77** [comment]
Sorry for my non-reactivity in my discussion, I was away from computers during this period.

Thanks for the changes, I'll take a closer look at how it works.

**Comment by textbrowser** [comment]
Welcomes return. Hope life be well.

**Reply by meteos77** [reply]
a new cat?

**Comment by meteos77** [comment]
To enable testing: add a new Z3950 server (BnF but with analytical record and Z3950) in biblioteq.conf

```
[Z39.50-9]

database_name = TOUT-ANA1-UTF8
format = utf-8
hostname = z3950.bnf.fr
name = BnF Z3950
password = Z3950_BNF
port = 2211
proxy_host =
proxy_port =
record_syntax = UNIMARC
timeout = 10
username = Z3950
```

**Reply by meteos77** [reply]
In my research, I came across this site, if it can be of help to other BiblioteQ users.
Each country has its own references for records.

[Z3950 & SRU servers](https://www.z-brary.com/)

**Comment by meteos77** [comment]
ISBN13 = 9791032404898
Z3950  result = 003 http://catalogue.bnf.fr/ark:/12148/cb46877916p
SRU result = ark:/12148/cb46877916p

**Reply by meteos77** [reply]
So there's a filtering difference between Z3950 and SRU.

If we want to keep only ark:// (short SRU format), we can use the cover image retrieval feature (Download / Others).
That's great!

**Reply by meteos77** [reply]
On the other hand, keeping the long format (without the 003) would make it possible to access the resource on the Internet if you can create this information in the form of a permanent Internet link.
003 http://catalogue.bnf.fr/ark:/12148/cb46877916p
in that case, you'd have to modify the functionality for covers.

**Reply by meteos77** [reply]
BiblioteQ also has a field called URL ; Is this the right place for this information?

**Reply by meteos77** [reply]
For me, the best version would be the short version as an alternative identifier (ark://) and the url version in the URL field (http://...).

After that, it's up to you to decide on the best choice (Development time / Difficulties / Interest).

**Comment by textbrowser** [comment]
Added BnF. Extract ARK string. Completed.

**Reply by meteos77** [reply]
Super, thank you, it works well with my tests

**Comment by textbrowser** [comment]
Fields can contain any information which they allow. BQ may override.

**Comment by textbrowser** [comment]
Even more contributions. Completed.
